static-assets/components/cstudio-forms/data-sources/img-CMIS-upload.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2007-2019 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+CStudioForms.Datasources.ImgCMISUpload = CStudioForms.Datasources.ImgCMISUpload ||
+function(id, form, properties, constraints)  {
+   	this.id = id;
+   	this.form = form;
+   	this.properties = properties;
+   	this.constraints = constraints;
+   	
+   	for(var i=0; i<properties.length; i++) {
+   		if(properties[i].name == "repoPath") {
+			this.repoPath = properties[i].value;
+		}
+		if(properties[i].name === "repositoryId") {
+			this.repositoryId = properties[i].value;
+		}
+   	} 
+	
+	return this;
+}
+
+YAHOO.extend(CStudioForms.Datasources.ImgCMISUpload, CStudioForms.CStudioFormDatasource, {
+	itemsAreContentReferences: true,
+
+    decreaseFormDialog: function(){
```

Comment:
Function is duplicate across various dialogs, share it. Same for other instances of this situation.

Reply:
Done